clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/LoggingTrackingStoreClient.java

```diff
@@ -0,0 +1,214 @@
+  public CompletableFuture<Map<K, V>> batchGet(Set<K> keys) throws VeniceClientException {
+    CompletableFuture<Map<K, V>> resultFuture = new CompletableFuture<>();
+    CompletableFuture<VeniceResponseMap<K, V>> streamingResultFuture = streamingBatchGet(keys);
+
+    streamingResultFuture.whenComplete((response, throwable) -> {
+      if (throwable != null) {
+        resultFuture.completeExceptionally(throwable);
+      } else if (!response.isFullResponse()) {
+        resultFuture.completeExceptionally(
+            new VeniceClientException(
+                "Received partial response, returned entry count: " + response.getTotalEntryCount()
+                    + ", and key count: " + keys.size()));
+      } else {
+        resultFuture.complete(response);
+      }
+    });
+    // We intentionally use stats for batch-get streaming since blocking impl of batch-get is deprecated.
+    return resultFuture;
+  }
+
+  @Override
+  public void streamingBatchGet(Set<K> keys, StreamingCallback<K, V> callback) throws VeniceClientException {
+    long preRequestTimeInNS = System.nanoTime();
+    super.streamingBatchGet(keys, new LoggingStreamingCallback<>(getStoreName(), callback, preRequestTimeInNS));
+  }
+
+  @Override
+  public CompletableFuture<VeniceResponseMap<K, V>> streamingBatchGet(Set<K> keys) throws VeniceClientException {
+    Map<K, V> resultMap = new VeniceConcurrentHashMap<>(keys.size());
+    Queue<K> nonExistingKeyList = new ConcurrentLinkedQueue<>();
+
+    VeniceResponseCompletableFuture<VeniceResponseMap<K, V>> resultFuture = new VeniceResponseCompletableFuture<>(
+        () -> new VeniceResponseMapImpl<>(resultMap, nonExistingKeyList, false),
+        keys.size(),
+        Optional.empty());
+    streamingBatchGet(keys, super.getStreamingCallback(keys, resultMap, nonExistingKeyList, resultFuture));
+    return resultFuture;
+  }
+
+  @Override
+  public void compute(
+      ComputeRequestWrapper computeRequestWrapper,
+      Set<K> keys,
+      Schema resultSchema,
+      StreamingCallback<K, ComputeGenericRecord> callback,
+      long preRequestTimeInNS) throws VeniceClientException {
+    super.compute(
+        computeRequestWrapper,
+        keys,
+        resultSchema,
+        new LoggingStreamingCallback<>(getStoreName(), callback, preRequestTimeInNS),
+        preRequestTimeInNS);
+  }
+
+  public static <T> BiFunction<? super T, Throwable, ? extends T> getLoggingCallback(
+      String storeName,
+      long startTimeInNS) {
+    return (T value, Throwable throwable) -> {
+      double latency = LatencyUtils.getElapsedTimeFromNSToMS(startTimeInNS);
+      if (throwable != null) {
+        handleUnhealthyRequest(storeName, throwable, latency);
+        handleStoreExceptionInternally(throwable);
+      } else {
+        if (latency > DEFAULT_SINGLE_GET_LOGGING_THRESHOLD_MS) {
+          String logMessage =
+              String.format("Receive high latency single get request for store: %s with latency: {}", storeName);
```

Comment:
If we put `latency` into the log message and use it to filter out duplicate messages, I think it may not work well, as latency has a big range..
Maybe we should use some fixed string + store name instead to filter out duplicate logging?

Reply:
Hmm, the latency number is not rendered into the logMessage yet, I think it is only filtering towards the logging template